index_search.go

```diff
@@ -11,74 +12,106 @@ const (
 	DefaultLimit int64 = 20
 )
 
+func (i Index) SearchRaw(query string, request *SearchRequest) (*json.RawMessage, error) {
+	resp := &json.RawMessage{}
+
+	if request.Limit == 0 {
+		request.Limit = DefaultLimit
+	}
+
+	searchPostRequestParams := searchPostRequestParams(query, request)
+
+	req := internalRequest{
+		endpoint:            "/indexes/" + i.UID + "/search",
+		method:              http.MethodPost,
+		contentType:         contentTypeJSON,
+		withRequest:         searchPostRequestParams,
+		withResponse:        resp,
+		acceptedStatusCodes: []int{http.StatusOK},
+		functionName:        "SearchRaw",
+	}
+
+	if err := i.client.executeRequest(req); err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
 func (i Index) Search(query string, request *SearchRequest) (*SearchResponse, error) {
 	resp := &SearchResponse{}
 
-	searchPostRequestParams := make(map[string]interface{}, 14)
-
 	if request.Limit == 0 {
 		request.Limit = DefaultLimit
 	}
 
+	searchPostRequestParams := searchPostRequestParams(query, request)
+
+	req := internalRequest{
+		endpoint:            "/indexes/" + i.UID + "/search",
+		method:              http.MethodPost,
+		contentType:         contentTypeJSON,
+		withRequest:         searchPostRequestParams,
+		withResponse:        resp,
+		acceptedStatusCodes: []int{http.StatusOK},
+		functionName:        "Search",
+	}
+
+	if err := i.client.executeRequest(req); err != nil {
+		return nil, err
+	}
```

Comment:
Maybe this method can call directly `SearchRaw` and Unmarshal the result in `SearchResponse` to avoid duplicate code, WDYT?

Reply:
Ah great point... I was essentially doing that in the unit test but not the actual implementation 😅 I'll take your suggestion and make that change